presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/InnerJoinPredicateUtils.java

```diff
@@ -0,0 +1,119 @@
+/*
```

Comment:
As I noted here: https://github.com/Teradata/presto/pull/555#discussion_r117824885
I left this as is until I here more from @sopel39 about what he envisions.